### PR TITLE
bpo-29707: Document that os.path.ismount() is not able to detect bind mounts.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue29707

### CODE REVIEW

File: Doc/library/os.path.rst [hunk 1]
```diff
@@ -283,10 +283,11 @@ the :mod:`glob` module.)
...
    Return ``True`` if pathname *path* is a :dfn:`mount point`: a point in a
    file system where a different file system has been mounted.  On POSIX, the
-   function checks whether *path*'s parent, :file:`path/..`, is on a different
-   device than *path*, or whether :file:`path/..` and *path* point to the same
+   function checks whether *path*'s parent, :file:`{path}/..`, is on a different
+   device than *path*, or whether :file:`{path}/..` and *path* point to the same
    i-node on the same device --- this should detect mount points for all Unix
-   and POSIX variants.  On Windows, a drive letter root and a share UNC are
+   and POSIX variants.  It is not able to detect bind mounts on the same
+   filesystem.  On Windows, a drive letter root and a share UNC are
```
Comment (eryksun) [inline Doc/library/os.path.rst]: For completeness, I would add "except when :file:\`{path}/..\` and \*path\* point to the same i-node". For example, `sudo mount --bind . mountpoint`.

Comment (serhiy-storchaka) [reply Doc/library/os.path.rst]: If it does not work in general case, but works in some particular case, and you can not identify this case, you still can not rely on the result. I do not think that it is worth to add more details.

Comment (eryksun) [reply Doc/library/os.path.rst]: Maybe weasel it as "reliably detect" so people know that `ismount()` can be true for some bind mounts on the same filesystem. 

Why does `ismount` check the inode values as opposed to special casing "/"? It doesn't help with a `chroot` environment, so what am I missing?

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no long accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

File: Doc/library/os.path.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/library/os.path.rst]: I suppose it is just easier on C. No need to normalize a path just to handle "//", "/.", "/..", "/.././", etc.

**Pull request merged**